### DISCUSSION TITLE
Relay power in bug?

### DISCUSSION THREAD

**Discussion by Barneor** [FakeFishGames/Barotrauma#14302] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

![image](https://github.com/user-attachments/assets/c4de4295-6f3b-4dbf-9d84-7695cb7b1414)

I think the relay should not transfer power between current sources connected to “Power_in”

### Reproduction steps

Go into the boat editor, spawn the power supply, connect it to the “Power in” relay. Then connect the consumer to the “Power in” relay. The consumer will start taking power from “Power in” although there is “Power out” for this purpose

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I'm not entirely sure I understand the issue correctly, do you mean that both the junction box and the relay are connected to the "power_in" input of the relay? In that case, power will pass through the circuit regardless if the relay is on or off. Devices connected to the same connection are essentially considered part of the same grid,

**Reply by Barneor** [reply]
Thanks now I see, I just thought that “Power in” will not allow to power other consumers, because in “Power in” the energy just goes into the relay, and “Power out” will actually allow to power the consumers.

The thing was that the relay with the input feeds the cable box.

I'm writing through a translator, sorry for the inaccuracy of the translation.

**Reply by Regalis11** [reply]
One way to think about it is that every device wired to the same connection is part of the same "pool" of power, and the relay only stops power from flowing from the power_in connection to the power_out connection.
![image](https://github.com/user-attachments/assets/01dda10a-aa57-457e-9516-a9c7a0151323)

I think you probably need a third relay to prevent the junction box from recharging from the reactor.

**Comment by SomeRandomNoobKekeke** [comment]
https://github.com/FakeFishGames/Barotrauma/issues/12696

As far as I know, this doesn't bypass the relay's power transfer limit, so I guess it's not a big problem
